<launch>
  <arg name="robot_name" default="turtlebot"/>
  <group ns="turtlebot" >
  <node name="exploration" output="screen" pkg="exploration" respawn="false" type="exploration">
    <param name="footprint_padding" value="0.01" />
    <param name="controller_frequency" value="10.0" />
    <param name="controller_patience" value="3.0" />

    <param name="oscillation_timeout" value="30.0" />
    <param name="oscillation_distance" value="0.5" />
    
    <rosparam file="$(find exploration)/config/move_base_gazebo/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find exploration)/config/move_base_gazebo/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find exploration)/config/move_base_gazebo/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find exploration)/config/move_base_gazebo/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find exploration)/config/move_base_gazebo/base_local_planner_params.yaml" command="load" /> 
    <rosparam file="$(find exploration)/config/move_base_gazebo/move_base_params.yaml" command="load" />

    <param name="global_costmap/robot_base_frame" value="/$(arg robot_name)/base_footprint"/>
    <param name="local_costmap/robot_base_frame" value="/$(arg robot_name)/base_footprint"/>
    <param name="local_costmap/global_frame" value="/$(arg robot_name)/odom"/>
    <param name="global_costmap/origin_x" value="-50"/>
    <param name="global_costmap/origin_y" value="-50"/>
    <param name="global_costmap/width" value="100" />
    <param name="global_costmap/height" value="100" />

    <remap from="map" to="/$(arg robot_name)/map" />
<!--   
    <param name="global_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint"/>
    <param name="local_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint"/>
    <param name="local_costmap/global_frame" value="$(arg robot_name)/odom"/>
    <param name="global_costmap/origin_x" value="-5"/>
    <param name="global_costmap/origin_y" value="-5"/>


    <param name="local_costmap/bump" value="$(arg robot_name)/sensors/bumper_pointcloud" />
    <param name="global_costmap/bump" value="$(arg robot_name)/sensors/bumper_pointcloud" />
-->
  </node>
</group>
</launch>